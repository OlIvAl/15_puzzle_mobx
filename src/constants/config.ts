export const BOARD_TILE_SIZE: number = 4;

export const TILE_SIZE: number = 100;
export const TILE_MARGIN: number = -1;
export const BORD_BORDER: number = 3;

export const BORD_SIZE: number = (TILE_SIZE * BOARD_TILE_SIZE) + (TILE_MARGIN * (BOARD_TILE_SIZE - 1));