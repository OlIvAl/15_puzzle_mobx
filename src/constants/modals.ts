export const WIN_MODAL = 'WIN_MODAL';