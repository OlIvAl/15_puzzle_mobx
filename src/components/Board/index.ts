import styled from 'styled-components';
import {BORD_BORDER, BORD_SIZE} from '../../constants/config';

const StyledBoard = styled.div`
  width: ${BORD_SIZE}px;
  height: ${BORD_SIZE}px;
  border: ${BORD_BORDER}px solid #aaa;
  background-color: #e5e5e5;
  position: relative;
  
  @media (max-width: ${BORD_SIZE * 1.2}px) {
    width: ${BORD_SIZE / 1.5}px;
    height: ${BORD_SIZE / 1.5}px;
  }
`;

export default StyledBoard;