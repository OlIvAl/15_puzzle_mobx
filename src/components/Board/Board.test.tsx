import * as React from 'react';
import Board from '.';
import {BORD_SIZE} from '../../constants/config';
import renderer from 'react-test-renderer';

describe('Board', () => {
  it('он отображается', () => {
    const tree = renderer.create(<Board />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('имеет заданную ширину', () => {
    const tree = renderer.create(<Board />).toJSON();

    expect(tree).toHaveStyleRule('width', `${BORD_SIZE}px`);
  });

  it('имеет заданную высоту', () => {
    const tree = renderer.create(<Board />).toJSON();

    expect(tree).toHaveStyleRule('height', `${BORD_SIZE}px`);
  });

  it('имеет заданную ширину в мобильном виде', () => {
    const tree = renderer.create(<Board />).toJSON();

    expect(tree).toHaveStyleRule(
      'width',
      `${BORD_SIZE / 1.5}px`,
      {
        media: `(max-width:${BORD_SIZE * 1.2}px)`
      }
    );
  });

  it('имеет заданную высоту в мобильном виде', () => {
    const tree = renderer.create(<Board />).toJSON();

    expect(tree).toHaveStyleRule(
      'height',
      `${BORD_SIZE / 1.5}px`,
      {
        media: `(max-width:${BORD_SIZE * 1.2}px)`
      }
    );
  });
});