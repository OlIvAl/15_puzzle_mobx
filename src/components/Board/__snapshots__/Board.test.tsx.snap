// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board он отображается 1`] = `
.c0 {
  width: 397px;
  height: 397px;
  border: 3px solid #aaa;
  background-color: #e5e5e5;
  position: relative;
}

@media (max-width:476.4px) {
  .c0 {
    width: 264.6666666666667px;
    height: 264.6666666666667px;
  }
}

<div
  className="c0"
/>
`;
